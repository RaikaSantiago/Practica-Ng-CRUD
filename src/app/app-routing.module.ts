import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
  {
    path: "usuarios",
    loadChildren: () =>
      import("./modules/usuario/usuario.module").then(
        (m) => m.UsuarioModule
      ),
  },
  { path: "", redirectTo: "/usuarios/list", pathMatch: "full" },
  { path: "**", redirectTo: "/usuarios/list", pathMatch: "full" },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
