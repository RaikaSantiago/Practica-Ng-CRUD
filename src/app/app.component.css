.icon-usermas{
    font-size: 20px;
}

.layout-main{
    padding-top: 100px !important;
}