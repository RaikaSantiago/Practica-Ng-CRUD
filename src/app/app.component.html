
<app-topbar></app-topbar>

<div class="container">
    <div class="layout-main">
        <router-outlet></router-outlet>
    </div>
</div>
<!-- Footer -->
<div class="text-center final-center">
    &copy; {{fechaActual}} Santiago Rivera Escobar | Full-Stack Developer
</div>