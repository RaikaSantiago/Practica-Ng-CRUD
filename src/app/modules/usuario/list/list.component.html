<ng-container *ngIf="loading">
    <div class="text-center jqbox_innerhtml">
        <div class="lds-ripple">
            <div></div>
            <div></div>
        </div>
    </div>
</ng-container>
<ng-container *ngIf="arrUsers.length === 0 && !loading">
    <div class="row">
        <div class="col-12">
            <div class="text-center">
                <h1>USUARIOS</h1>
            </div>
        </div>
    </div>
    <div class="row">
        <div class="col-12">
            <div class="text-center">
                <div>
                    <img src="./assets/images/79572-empty-state.gif" height="300" alt="" />
                </div>
                <h5>No se encontraron usuarios para esta consulta, desea agregar un usuario ?</h5>
                <br>
                <a (click)="open(content)" style="cursor: pointer;" tooltip="Agregar Usuario" placement="top"
                    showDelay="500" class="ov-btn-slide-left">
                    <i class="fa fa-user-plus icon-usermas" aria-hidden="true"></i>
                </a>
            </div>
        </div>
    </div>
</ng-container>
<ng-container *ngIf="arrUsers.length > 0 && !loading">
    <div class="row">
        <div class="col-12">
            <div class="text-center">
                <h1 class="h1-vol">USUARIOS</h1>
            </div>
        </div>
    </div>
    <div class="row">
        <div class="col-12">
            <div class="table-responsive">
                <table class="table">
                    <thead class="thead-dark">
                        <tr>
                            <th scope="col">#</th>
                            <th scope="col">CÃ‰DULA</th>
                            <th scope="col">NOMBRES</th>
                            <th scope="col">APELLIDOS</th>
                            <th scope="col">TELEFONO</th>
                            <th scope="col">CORREO</th>
                            <th class="text-center" scope="col">ACCIONES</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr *ngFor="let user of arrUsers | slice: (page -1) * pageSize : page * pageSize; let i=index ">
                            <th scope="row">{{i + 1}}</th>
                            <td>{{user.cedula}}</td>
                            <td>{{user.nombres}}</td>
                            <td>{{user.apellidos}}</td>
                            <td>{{user.telefono}}</td>
                            <td>{{user.correo}}</td>
                            <td>
                                <div class="text-center">
                                    <button (click)="open(content, user)" type="button"
                                        style="font-size:small;margin-right: 3px;" tooltip="Editar Registro"
                                        placement="top" showDelay="500" class="btn btn-primary"><i class="fa fa-pencil"
                                            aria-hidden="true"></i></button>
                                    <button (click)="deleteR(user.id)" type="button" style="font-size: small;"
                                        tooltip="Eliminar Registro" placement="top" showDelay="500"
                                        class="btn btn-danger"><i class="fa fa-trash" aria-hidden="true"></i></button>
                                </div>
                            </td>
                        </tr>
                    </tbody>
                </table>
                <ngb-pagination class="d-flex justify-content-center" [(page)]="page" [pageSize]="pageSize" [collectionSize]="arrUsers.length"></ngb-pagination>
            </div>
        </div>
    </div>
    <div class="row">
        <div class="col-12">
            <div class="text-center">
                <a style="cursor: pointer;" (click)="open(content)" tooltip="Agregar Usuario" placement="top"
                    showDelay="500" class="ov-btn-slide-left">
                    <i class="fa fa-user-plus icon-usermas" aria-hidden="true"></i>
                </a>
            </div>
        </div>
    </div>
</ng-container>

<ng-template #content let-dismiss="dismiss">
    <app-form [data]="dataSelect" [dismiss]="dismiss"></app-form>
</ng-template>