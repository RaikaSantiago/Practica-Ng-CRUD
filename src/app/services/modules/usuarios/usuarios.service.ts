import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../../../environments/environment';
import { ResponseMessageModel, UsuarioModel } from 'src/app/models/usuario.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UsuariosService {
  private headers: any;

  constructor(public http: HttpClient) {
    this.create_headers();
  }

  private create_headers() {
    this.headers = new HttpHeaders();
    this.headers.set('Content-type', 'application/json');
    this.headers.set('Access-Control-Allow-Origin', '*');
  }

  getObtenerUsuarios(module: string): Observable<UsuarioModel[]> {
    this.create_headers();
    return this.http.get<UsuarioModel[]>(`${environment.urlGlobal}/${module}/get`, { headers: this.headers });
  }

  getBuscarUsuarioId(module: string, id: number): Observable<UsuarioModel> {
    this.create_headers();
    return this.http.get<UsuarioModel>(`${environment.urlGlobal}/${module}/search/${id}`, { headers: this.headers });
  }

  putActualizarUsuario(module: string, body: any): Observable<ResponseMessageModel> {
    return this.http.put<ResponseMessageModel>(`${environment.urlGlobal}/${module}/put`, body, { headers: this.headers });
  }

  postGuardarUsuario(module: string, body: any): Observable<ResponseMessageModel> {
    return this.http.post<ResponseMessageModel>(`${environment.urlGlobal}/${module}/post`, body, { headers: this.headers });
  }

  deleteEliminarUsuario(module: string, id: number): Observable<ResponseMessageModel> {
    return this.http.delete<ResponseMessageModel>(`${environment.urlGlobal}/${module}/delete/${id}`, { headers: this.headers });
  }

}
