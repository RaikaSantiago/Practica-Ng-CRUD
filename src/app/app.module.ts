import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { UsuarioModule } from './modules/usuario/usuario.module';
import { UtilModule } from './util/util.module';
import { GlobalModule } from './modules/global.module';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    UtilModule,
    UsuarioModule,
    GlobalModule,
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
