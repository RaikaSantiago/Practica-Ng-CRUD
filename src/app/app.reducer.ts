import { ActionReducerMap } from "@ngrx/store";
import { MetaReducer } from "@ngrx/store";
import { hydrationMetaReducer } from "./ngrx/store/reducers/hydrations.reducer";
import { usuarioReducer, UsuarioState } from "./ngrx/store/reducers/usuarios.reducer";


export interface AppState {
    usuarios: UsuarioState;
}

export const appReducers: ActionReducerMap<AppState> = {
    usuarios: usuarioReducer,
};

export const metaReducers: MetaReducer<any>[] = [hydrationMetaReducer];


