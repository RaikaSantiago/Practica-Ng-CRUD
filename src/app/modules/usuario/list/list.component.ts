import { Component, OnInit } from '@angular/core';
import { NgbModal, NgbModalConfig } from '@ng-bootstrap/ng-bootstrap';
import { UsuarioModel } from 'src/app/models/usuario.model';
import Swal from 'sweetalert2';
import { UsuariosService } from '../../../services/modules/usuarios/usuarios.service';
import { ResponseMessageModel } from '../../../models/usuario.model';
import { AppState } from 'src/app/app.reducer';
import { Store } from '@ngrx/store';
import * as usuariosActions from '../../../ngrx/store/actions/usuarios.actions';

@Component({
  selector: 'app-list',
  templateUrl: './list.component.html',
  styleUrls: ['./list.component.css']
})
export class ListComponent implements OnInit {

  page: number = 1;
  pageSize: number = 3;
  arrUsers: UsuarioModel[] = [];
  loading: boolean = true;
  dataSelect: UsuarioModel;

  constructor(private _usuarioService: UsuariosService,
    public config: NgbModalConfig,
    private store: Store<AppState>,
    private modalService: NgbModal) {
    config.size = 'lg';
  }

  async ngOnInit() {
    await this.consultaUsuarios();
    this.selectStore();
  }

  async consultaUsuarios() {
    await this._usuarioService.getObtenerUsuarios('usuario').toPromise().then(res => {
      setTimeout(() => {
        this.store.dispatch(usuariosActions.addUsuarios({
          list: res
        }));
        this.loading = false;
      }, 700);
    }, err => {
      console.log(err);
      this.loading = false;
    })
  }

  selectStore() {
    this.store.select('usuarios').subscribe(res => {
      this.arrUsers = [...res.list]
    })
  }

  open(content: any, data?: UsuarioModel) {
    this.dataSelect = data;
    this.modalService.open(content);
  }

  deleteR(id: number) {
    Swal.fire({
      title: 'Estas seguro?',
      text: "No podrÃ¡s revertir esto!",
      icon: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Si, Eliminar!'
    }).then(async (result) => {
      if (result.isConfirmed) {
        await this._usuarioService.deleteEliminarUsuario('usuario', id).toPromise().then((res: ResponseMessageModel) => {
          if (res.status === 200) {
            Swal.fire(
              'Eliminado!',
              'El registro ha sido eliminado.',
              'success'
            )
            this.store.dispatch(usuariosActions.deleteUsuario({id}));
          }
        }, err => {
          console.error(err);
          Swal.fire({
            icon: 'error',
            title: 'Fallo',
            text: 'Se presento un error inesperado al eliminar el registro.',
            showConfirmButton: true,
          })
        })
      }
    })
  }

}
