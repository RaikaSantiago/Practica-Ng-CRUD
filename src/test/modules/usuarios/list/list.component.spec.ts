import { ComponentFixture, TestBed } from '@angular/core/testing';

import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { StoreModule } from '@ngrx/store';
import { appReducers, metaReducers } from '../../../../app/app.reducer';
import { ListComponent } from '../../../../app/modules/usuario/list/list.component';
import { UsuariosService } from '../../../../app/services/modules/usuarios/usuarios.service';

describe('ListComponent', () => {
  let component: ListComponent;
  let fixture: ComponentFixture<ListComponent>;
  let compiled: HTMLElement;
  let service: UsuariosService;
  let httoMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ListComponent],
      imports: [
        HttpClientTestingModule,
        StoreModule.forRoot(appReducers, { metaReducers })
      ],
      providers: [
        UsuariosService
      ]
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ListComponent);
    component = fixture.componentInstance;
    service = TestBed.inject(UsuariosService);
    httoMock = TestBed.inject(HttpTestingController);

    fixture.detectChanges();
    compiled = fixture.nativeElement as HTMLElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('Debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  })

  test('Debe contener el method GET en la consulta usuarios', () => {
    const request = httoMock.expectOne('http://localhost:8080/usuario/get');
    expect(request.request.method).toBe('GET');
  })

});
