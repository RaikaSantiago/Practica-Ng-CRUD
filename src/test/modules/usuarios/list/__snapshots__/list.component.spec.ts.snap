// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent Debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="text-center jqbox_innerhtml"
  >
    <div
      class="lds-ripple"
    >
      <div />
      <div />
    </div>
  </div>
  
  
  
  
  
</div>
`;
